src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_SIZE = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
```

Comment:
the method is too complicated. Simplify it. 

1. resize if need 
2. get index by hash 
3. Check bucket by index, if it == null  then put new node 
4. else: 5 and 6
5. go thru all elements and check them on the same key, if find - just rewrite the value, and call return  
6. otherwise - add the node to the end of the list. 
7. after if-else change size.